OpenRA.Mods.Common/AI/HackyAI.cs

```diff
@@ -22,6 +22,23 @@
 
 namespace OpenRA.Mods.Common.AI
 {
+	class CapturerTarget<TInfoType> where TInfoType : class, ITraitInfoInterface
```

Comment:
IMO you wouldn't lose any generality by renaming it.  The current name feels rather awkward, which then implies (to me at least) that you might be using it in an awkward way when you're actually not.